assignment1/class.dart

```diff
@@ -0,0 +1,91 @@
+enum Type { raw, manufactured, imported }
+
+abstract class Item {
+  static List<dynamic> currentItems = [];
+
+  double priceCalculation();
+
+  factory Item(
+      Type itemType, String itemName, int itemPrice, int itemQuantity) {
+    switch (itemType) {
+      case Type.imported:
+        return importedItem(
+            itemName: itemName,
+            itemPrice: itemPrice,
+            itemQuantity: itemQuantity);
+
+      case Type.manufactured:
+        return manufacturedItem(
+            itemName: itemName,
+            itemPrice: itemPrice,
+            itemQuantity: itemQuantity);
+      case Type.raw:
+        return rawItem(
+            itemName: itemName,
+            itemPrice: itemPrice,
+            itemQuantity: itemQuantity);
+      default:
+        return rawItem(
+            itemName: itemName,
+            itemPrice: itemPrice,
+            itemQuantity: itemQuantity);
+    }
+  }
+}
+
+class importedItem implements Item {
+  int itemPrice;
+  String itemName;
+  int itemQuantity;
+  importedItem(
+      {required this.itemName,
+      required this.itemPrice,
+      required this.itemQuantity});
+
+  @override
+  double priceCalculation() {
```

Comment:
Method names wording can be improved to denote the action. `calculatePrice()` might be a better name here